pkg/internal/controller/controller.go

```diff
@@ -171,41 +173,55 @@ func (c *Controller[request]) Start(ctx context.Context) error {
 		// NB(directxman12): launch the sources *before* trying to wait for the
 		// caches to sync so that they have a chance to register their intendeded
 		// caches.
+		errGroup, _ := errgroup.WithContext(ctx)
```

Comment:
nit: If you don't need the context, can you just use:
```suggestion
		errGroup := new(errgroup.Group)
```